examples/pissa_finetuning/pissa_finetuning.py

```diff
@@ -0,0 +1,129 @@
+import torch
+import os
+from peft import LoraConfig, get_peft_model, PeftModel, prepare_model_for_kbit_training
+from dataclasses import dataclass, field
+from typing import Optional, Union, List
+from transformers import AutoTokenizer, AutoModelForCausalLM, BitsAndBytesConfig, TrainingArguments, HfArgumentParser
+from peft.utils.pissa_utils import pissa_pre_training_saving, pissa_post_training_saving
+from trl import SFTTrainer
+from datasets import load_dataset
+
+
+@dataclass
+class TrainingArguments(TrainingArguments):
+    # model configs
+    base_model_name_or_path: Optional[str] = field(
+        default="meta-llama/Llama-2-7b-hf", metadata={"help": "The name or path of the fp32/16 base model."}
+    )
+    residual_model_name_or_path: Optional[str] = field(
+        default=None,
+        metadata={
+            "help": "The name or path of the fp32/16 residual model. (`['fxmeng/pissa-llama-2-7b-r16-alpha-16']`)"
+        },
+    )
+    bits: int = field(default=4, metadata={"help": "(`['4', '8', '16']`)"})
```

Comment:
Does it really make sense to have `16` as an option here? IIUC, if bits != 4 or 8, it just means no quantization (so it could also be float32). How about making this `Optional[int]`, with `None` meaning no quant?